### DISCUSSION TITLE
Application not working with generated bearer token from postman

### DISCUSSION THREAD

**Discussion by hussainshahzad5471** [iTwin/community#44] [General]
I am able to get a bearer token through javascript, but when I use that token I get
_{"code":"IModelNotFound","message":"Requested iModelId is not available."}_
somehow the javascript generated bearer token dosent work on the UI but the one  generated from the UI works. 

Below is the javascript to generate Bearer Token 
```ts
const myHeaders = new Headers();
  myHeaders.append("Content-Type", "application/x-www-form-urlencoded");
  myHeaders.append("Cookie", "PF=tAYsumtPXGiXR1LZgD50mV; cchallenge=true; idfirst_redirect_uri=http://localhost:3000/signin-oidc; JSESSIONID=node0mhp0bbmlghk4pmcdn1gk2swi3261669.node0");
  const urlencoded = new URLSearchParams();
  urlencoded.append("grant_type", "client_credentials");
  urlencoded.append("client_id", "service-oueyPNpUb0gKZMAMhHQV31SX4");
  urlencoded.append("client_secret", {YOUR_SECRET});
  //urlencoded.append("scope", "imodelaccess:read");
  urlencoded.append("scope", "imodelaccess:read imodels:read realitydata:read itwin-platform");
  const requestOptions = {
    method: "POST",
    headers: myHeaders,
    body: urlencoded,
    redirect: "follow"
  };
  const res = await fetch("https://ims.bentley.com/connect/token", requestOptions)
    .then((response) => response.text())
    .then((result) => {token = result.toString()})
    .catch((error) => console.error(error));
```

**Comment by aruniverse** [comment]
If you're using client credentials, have you added the service client as a member of the itwin you are trying to access?

**Comment by hussainshahzad5471** [comment]
Hi Arun,
I didn't find anything on the UI (https://developer.bentley.com/my-itwins/) to add the service client as a member of the iTwin, please guide me how I could do that. FYI, I am still on the free trial.

**Reply by aruniverse** [reply]
You can use the manage members participants option to add your service client as a member of the itwin.
The service client's email will be something like ` service-{SOME_GUID_HERE}@apps.imsoidc.bentley.com`
<img width="451" alt="image" src="https://github.com/iTwin/community/assets/11051042/19f93f04-3bf1-4a74-a0b4-e5cc1087f3dd">

**Reply by hussainshahzad5471** [reply]
Thanks, it works now!!
